### PR TITLE
Cleanup mixlib-authentication

### PR DESCRIPTION
**Pull request by jaym**
This cleanup does a couple things:
- Updates the Rakefile to work with current versions of rake
- Add a gemfile
- Specify test dependencies in gemspec
- Update rspec tests to use expect syntax
- Add a version.rb
- Add travis.yml

### CODE REVIEW

**Comment by ksubrama** [comment]
:+1:

**Comment by danielsdeleo** [comment]
Probably want the travis yaml to restrict builds to pull requests and master so it doesn't double-test PRs.

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+source "https://rubygems.org"
+gemspec
+
+gem 'rake'
```
Comment (danielsdeleo) [inline Gemfile]: should this be a development dependency in the gemspec instead?

**Comment by danielsdeleo** [comment]
:+1:

**Pull request merged**